packages/server/test/utils/xdccDatabase.test.ts

```diff
@@ -1,79 +1,219 @@
+			mockFetch.mockResolvedValue({
+				text: () => Promise.resolve(""),
+			} as Response);
+
+			const database = await parse();
+
+			expect(database).toStrictEqual([]);
+		});
+
+		it("should handle only header lines", async () => {
+			mockFetch.mockResolvedValue({
+				text: () => Promise.resolve("[section1]\n[section2]"),
+			} as Response);
+
+			const database = await parse();
+
+			expect(database).toStrictEqual([]);
+		});
 	});
 
 	describe("create", () => {
-		it("should create a searchable orama database", async () => {
-			globalThis.fetch = vi.fn().mockResolvedValue({
-				text: () => `#1   AeC|1P|01 1.9G Foo-Fuugther.part01.rar
-#2   AeC|1P|01 1.9G Foo-Fuugther.part02.rar`,
-			});
+		it("should create database and populate with channel data", async () => {
+			mockFetch.mockResolvedValue({
+				text: () =>
+					Promise.resolve(`#1   AeC|1P|01 1.9G Foo-Fuugther.part01.rar
+#2   AeC|1P|01 1.9G Foo-Fuugther.part02.rar`),
+			} as Response);
 
 			await create(parsedXdccDatabase);
-			const result = await search("foo");
 
-			expect(result).toStrictEqual([
-				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part01.rar",
-					fileNumber: "#1",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#a-b-c",
-				},
+			expect(mockSqlite3.Database).toHaveBeenCalledWith(":memory:");
+			expect(mockPrepare).toHaveBeenCalledWith("INSERT INTO files VALUES (?, ?, ?, ?, ?, ?)");
+		});
+
+		it("should filter out invalid script lines", async () => {
+			mockFetch.mockResolvedValue({
+				text: () =>
+					Promise.resolve(`#1   Bot1 100M Valid File.txt
+invalid line
+#2
+   
+#3   Bot2 200M Another Valid.txt`),
+			} as Response);
+
+			await create([
 				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part02.rar",
-					fileNumber: "#2",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#a-b-c",
+					channelName: "#test",
+					network: "irc.test.net",
+					scriptUrl: "http://test.com/script.php",
 				},
+			]);
+
+			expect(mockPrepare).toHaveBeenCalled();
+		});
+	});
+
+	describe("search", () => {
+		it("should search by filename", async () => {
+			const testResults = [
 				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part01.rar",
+					channelName: "#test-channel",
+					network: "irc.test.net",
 					fileNumber: "#1",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#Pierpaolo",
-				},
-				{
-					channelName: "AeC|1P|01",
-					fileName: "Foo-Fuugther.part02.rar",
-					fileNumber: "#2",
-					fileSize: "1.9G",
-					network: "irc.foo.biz",
-					serverName: "#Pierpaolo",
+					botName: "TestBot",
+					fileSize: "500M",
+					fileName: "Some File Name.txt",
 				},
-			]);
+			];
+
+			mockAll.mockImplementation((_query, _params, callback) => {
+				callback(null, testResults);
+			});
+
+			const result = await search("some name");
+
+			expect(result).toHaveLength(1);
+			expect(result[0]).toMatchObject({
```

Comment:
check the whole result object with a single strict assertion.
apply the same for all the other tests